patterns/ecs-v1/linux-syslog

```diff
@@ -1,16 +1,16 @@
 SYSLOG5424PRINTASCII [!-~]+
 
-SYSLOGBASE2 (?:%{SYSLOGTIMESTAMP:timestamp}|%{TIMESTAMP_ISO8601:timestamp8601}) (?:%{SYSLOGFACILITY} )?%{SYSLOGHOST:logsource}+(?: %{SYSLOGPROG}:|)
-SYSLOGPAMSESSION %{SYSLOGBASE} (?=%{GREEDYDATA:message})%{WORD:pam_module}\(%{DATA:pam_caller}\): session %{WORD:pam_session_state} for user %{USERNAME:username}(?: by %{GREEDYDATA:pam_by})?
+SYSLOGBASE2 (?:%{SYSLOGTIMESTAMP:timestamp}|%{TIMESTAMP_ISO8601:timestamp}) (?:%{SYSLOGFACILITY} )?%{SYSLOGHOST:[host][hostname]}+(?: %{SYSLOGPROG}:|)
+SYSLOGPAMSESSION %{SYSLOGBASE} (?=%{GREEDYDATA:message})%{WORD:[system][auth][pam][module]}\(%{DATA:[system][auth][pam][origin]}\): session %{WORD:[system][auth][pam][session_state]} for user %{USERNAME:[user][name]}(?: by %{GREEDYDATA})?
 
 CRON_ACTION [A-Z ]+
-CRONLOG %{SYSLOGBASE} \(%{USER:user}\) %{CRON_ACTION:action} \(%{DATA:message}\)
+CRONLOG %{SYSLOGBASE} \(%{USER:[user][name]}\) %{CRON_ACTION:[system][cron][action]} \(%{DATA:message}\)
 
 SYSLOGLINE %{SYSLOGBASE2} %{GREEDYDATA:message}
 
 # IETF 5424 syslog(8) format (see http://www.rfc-editor.org/info/rfc5424)
-SYSLOG5424PRI <%{NONNEGINT:syslog5424_pri}>
+SYSLOG5424PRI <%{NONNEGINT:[log][syslog][facility][code]}>
 SYSLOG5424SD \[%{DATA}\]+
-SYSLOG5424BASE %{SYSLOG5424PRI}%{NONNEGINT:syslog5424_ver} +(?:%{TIMESTAMP_ISO8601:syslog5424_ts}|-) +(?:%{IPORHOST:syslog5424_host}|-) +(-|%{SYSLOG5424PRINTASCII:syslog5424_app}) +(-|%{SYSLOG5424PRINTASCII:syslog5424_proc}) +(-|%{SYSLOG5424PRINTASCII:syslog5424_msgid}) +(?:%{SYSLOG5424SD:syslog5424_sd}|-|)
+SYSLOG5424BASE %{SYSLOG5424PRI}%{NONNEGINT:[system][syslog][version]} +(?:%{TIMESTAMP_ISO8601:timestamp}|-) +(?:%{IPORHOST:[host][hostname]}|-) +(-|%{SYSLOG5424PRINTASCII:[system][syslog][app_name]}) +(-|%{SYSLOG5424PRINTASCII:[system][syslog][procid]}) +(-|%{SYSLOG5424PRINTASCII:[system][syslog][msgid]}) +(?:%{SYSLOG5424SD:[system][syslog][structured_data]}|-|)
 
-SYSLOG5424LINE %{SYSLOG5424BASE} +%{GREEDYDATA:syslog5424_msg}
+SYSLOG5424LINE %{SYSLOG5424BASE} +%{GREEDYDATA:message}
```

Comment:
this is a bit of a 'breaking' change but we likely want all syslog logs to match in a similar way, don't we?

as others (`SYSLOGLINE`) do append to `event["message"]` ... 

an alternative (still breaking backward compat for `SYSLOGLINE`) would be for the message part to always go to `system.syslog.message` (or `system.cron.message`) and allow for a clean rename afterwards - `SYSLOGLINE` sample : 

`Mar 16 00:01:25 evita postfix/smtpd[1713]: connect from camomile.cloud9.net[168.100.1.3]`
```
  "timestamp"=>"Mar 16 00:01:25",
  "logsource"=>"evita",
  "program"=>"postfix/smtpd", 
  "pid"=>"1713", 
  "message"=>[
    "Mar 16 00:01:25 evita postfix/smtpd[1713]: connect from camomile.cloud9.net[168.100.1.3]", 
    "connect from camomile.cloud9.net[168.100.1.3]"
  ]
```
an alternative ecs proposal would be to not use `message` and always prefix extracted message sub-part : 
```
  "timestamp"=>"Mar 16 00:01:25", 
  "host"=>{"hostname"=>"evita"}
  "process"=>{"name"=>"postfix/smtpd", "pid"=>"1713"},
  "message"=>"Mar 16 00:01:25 evita postfix/smtpd[1713]: connect from camomile.cloud9.net[168.100.1.3]", 
  "system"=>{
    "syslog"=>{
      "message"=>"connect from camomile.cloud9.net[168.100.1.3]"
    }
  }
```

Reply:
Thanks Mat, we definitely want to handle all of that (the syslog input already does most of it except for keeping the original message around).

The discussion I was looking for here is whether LS wants to keep append-ing `message` as its matched,
or rather starts using name-spaced matching (striped out part would go to `system.syslog.message`) as a convention to keep the event structure easier to process.

Reply:
for the record: convention at this point is that we keep using (append-in to) the `message` field